api/server/models/user.js

```diff
@@ -20,80 +19,75 @@ module.exports = function (User) {
                         var resultSize = lodash.size(result);
                         if (resultSize > 0) {
                             result = result[0];
-
                             var r_verification_status = result.verification_status;
-
-                            if (action_type == 'facebook' || action_type == 'google') {
-                                var data = {
-                                    user_id: result.id
-                                };
-                                callback(null, 1, 'Success login', data);
-                            } else if (action_type == 'manual_register') {
+                            if (action_type == 'manual_register') {
                                 callback(null, 0, 'Email id already exists.', data);
-                            } else if (action_type == 'manual_login') {
-                                var exist_password = result.password;
-                                UTIL.match_encode_password(password, exist_password, function (check) {
-                                    if (check == true) {
-                                        if (r_verification_status == 0) {
-                                            callback(null, 0, 'Please verify you account first', {});
-                                        } else {
+                            } else if (action_type == 'manual_login' || action_type == 'facebook' || action_type == 'google') {
+                                if (action_type == 'facebook' || action_type == 'google') {
+                                    //password = "";
+                                }
+                                if (r_verification_status == 0) {
+                                    callback(null, 0, 'Please verify you account first', {});
+                                } else {
+                                    //-START--get access token---------
+                                    User.login({
+                                        email:email_id,
+                                        password : password
+                                    },function( err, accessToken ){
+                                        if( err ){
+                                            callback(null, 0, 'Invalid login', {});
+                                        }else{
                                             var data = {
-                                                user_id: result.id
+                                                user_id: result.id,
```

Comment:
do we need to send back user_id to frontend? will it be needed anywhere or any api